### PR TITLE
SSCS-10245 Use VenueDetails to find hearing location epims id.

### PR DESCRIPTION
**Pull request by dattipoeHMCTS**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by rmurphy-hmcts — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMapping.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        HearingLocations hearingLocations = new HearingLocations();
+        hearingLocations.setLocationId(epimsId);
+        hearingLocations.setLocationType("court");
+
+        List<HearingLocations> hearingLocationsList = new ArrayList<>();
+        hearingLocationsList.add(hearingLocations);
+
+        return hearingLocationsList;
```
Comment (nickwest1) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMapping.java:40]: return List.of(hearingLocations);

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMapping.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+        VenueDetails venueDetails = referenceData.getVenueDataLoader().getVenueDetailsMap()
+            .get(String.valueOf(venueId));
+
+        if (venueDetails != null) {
+            epimsId = venueDetails.getEpimsId();
+        }
+
+        HearingLocations hearingLocations = new HearingLocations();
+        hearingLocations.setLocationId(epimsId);
```
Comment (nickwest1) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMapping.java:34]: hearingLocations.setLocationId(Optional.ofNullable(venueDetails).map(vd->getEpimsId); would be far cleaner

**Review by nickwest1 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMapping.java (hunk 2 above)
Comment (wmarshid) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMapping.java:34]: emm that's debatable.. the way it is currently looks more readable to me.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/ReferenceData.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.hmcts.reform.sscs.service;
+
+import lombok.Getter;
+import org.springframework.stereotype.Component;
+
+@Component
+@Getter
+public class ReferenceData {
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/ReferenceData.java:8]: Optional: Could we name this `VenueReferenceData`?

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMappingTest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+        List<HearingLocations> result = HearingLocationMapping.getHearingLocations(caseData, referenceData);
+
+        assertThat(result).hasSize(1);
+        assertThat(result.get(0).getLocationId()).isEqualTo("9876");
+        assertThat(result.get(0).getLocationType()).isEqualTo("court");
+    }
+
+    private void setupVenueMaps() {
+        Map<String, Integer> venueIdMap = new HashMap<>();
```
Comment (wmarshid) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMappingTest.java:58]: Optional: Can use `Map.of()` for these instead if you want.. like so;

```
Map<String, Integer> venueIdMap = Map.of(
            PROCESSING_VENUE_1, 68,
            PROCESSING_VENUE_2, 2);
```
            
```
       Map<String, VenueDetails> venueDetailsMap = Map.of(
            "68", VenueDetails.builder().epimsId("9876").build(),
            "2", VenueDetails.builder().epimsId("1111").build());
```

Sorry formatting is a bit weird.

**Review by wmarshid — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/service/ReferenceData.java (hunk 3 above)
Comment (dattipoeHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/ReferenceData.java:8]: There are going to be more classes in here

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMappingTest.java (hunk 4 above)
Comment (dattipoeHMCTS) [reply src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMappingTest.java:58]: My lack of J9+ is showing!

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMapping.java (hunk 1 above)
Comment (TeaWithLucas) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMapping.java:40]: I think this is where multiple locations will be added in 192, so I think that's the reason for creating a empty list?

File: src/main/java/uk/gov/hmcts/reform/sscs/service/ReferenceData.java (hunk 3 above)
Comment (TeaWithLucas) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/ReferenceData.java:8]: Yeah, my 116 classes

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMappingTest.java (hunk 4 above)
Comment (TeaWithLucas) [reply src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMappingTest.java]: .of is really nice!

**Review by TeaWithLucas — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMapping.java [hunk 5]
```diff
@@ -26,18 +25,15 @@ public static List<HearingLocations> getHearingLocations(SscsCaseData caseData,
         VenueDetails venueDetails = referenceData.getVenueDataLoader().getVenueDetailsMap()
             .get(String.valueOf(venueId));
 
-        if (venueDetails != null) {
-            epimsId = venueDetails.getEpimsId();
-        }
+        String epimsId = Optional.ofNullable(venueDetails)
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMapping.java:28]: I guess it's a compromise lol.. could maybe just use a ternary operator on line 33 if @nickwest1 is happy with that?

**Review by nickwest1 — DISMISSED** [review]

**Review by wmarshid — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/service/ReferenceData.java (hunk 3 above)
Comment (wmarshid) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/ReferenceData.java:8]: Okay.. please no @TeaWithLucas I can't take any more.

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMapping.java (hunk 1 above)
Comment (wmarshid) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMapping.java]: @TeaWithLucas makes sense I guess

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMapping.java (hunk 5 above)
Comment (nickwest1) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingLocationMapping.java:28]: I prefer what Del's done TBH

File: src/main/java/uk/gov/hmcts/reform/sscs/service/ReferenceData.java (hunk 3 above)
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/ReferenceData.java:8]: Optional/Minor: Should we name this reference data service?

**Review by wmarshid — CHANGES_REQUESTED** [review]

**Review by nickwest1 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java [hunk 6]
```diff
@@ -150,6 +151,20 @@ public static boolean shouldBeHearingsInWelshFlag() {
...
+                                                             ReferenceDataServiceHolder referenceDataServiceHolder) {
+
+        String epimsId = referenceDataServiceHolder
+            .getVenueService()
+            .getEpimsIdForVenue(processingVenue)
+            .orElse(null);
+
+        HearingLocations hearingLocation = new HearingLocations();
+        hearingLocation.setLocationId(epimsId);
+        hearingLocation.setLocationType("court");
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java:164]: Minor: magic string.. can we make this a constant?

**Review by wmarshid — APPROVED** [review]

**Review by nickwest1 — APPROVED** [review]

**Review by nickwest1 — DISMISSED** [review]

**Review by nickwest1 — DISMISSED** [review]

**Review by nickwest1 — APPROVED** [review]

**Review by endamccormack — APPROVED** [review]

**Pull request closed without merging**